add ability to go offline

djih opened the pull request:
Similar to: https://github.com/amplitude/Amplitude-Android/blob/master/src/com/amplitude/api/AmplitudeClient.java#L223-L232

curtisliu commented on AmplitudeTests/AmplitudeTests.m:
```diff
@@ -368,4 +368,30 @@ -(void)testAutoIncrementSequenceNumber {
     }
 }
 
+-(void)testSetOffline {
```
can you add a test that events still get truncated when offline?

djih replied:
added test and found a bug!

djih commented:
Fixed bug where events weren't being dropped if eventMaxCount < eventRemoveBatchSize (since getting nthEventId for eventRemoveBatchSize would return -1). Added tests for truncating events and truncating events while offline. PTAL @curtisliu

curtisliu commented on Amplitude/Amplitude.m:
```diff
@@ -533,6 +531,20 @@ - (void)logEvent:(NSString*) eventType withEventProperties:(NSDictionary*) event
+- (void)truncateEventQueues
+{
+    AMPDatabaseHelper *dbHelper = [AMPDatabaseHelper getDatabaseHelper];
+    int numEventsToRemove = self.eventMaxCount > kAMPEventRemoveBatchSize ? kAMPEventRemoveBatchSize : self.eventMaxCount/2;
```
how did you choose self.eventMaxCount / 2?

curtisliu replied:
I think this should be more like min(max(1, self.eventMaxCount / 10), kAMPEventRemoveBatchSize)

djih replied:
Changed

curtisliu commented:
LGTM
